### ISSUE TITLE
Calling wp_calculate_image_srcset in a multisite environment returns the wrong srcset

### ISSUE THREAD

**Issue by dominicpedro** [cloudinary/cloudinary_wordpress#718]
HI Support

I came across this issue after installing Cloudinary on my multi site. When calling `wp_calculate_image_srcset` from a template or functions file and the image is NOT synced with cloudinary, the srcset returned has a double path to the file.

eg: 
for a file:     https://tcs000000.local/africafoundation/wp-content/uploads/sites/6/HEALTH_nurse_blood-pressure-e1627651075100.jpg

the generated srcset is: 
https://tcs000000.local/africafoundation/wp-content/uploads/sites/6/sites/6/HEALTH_nurse_blood-pressure-e1627651075100-300x169.jpg 300w, https://tcs000000.local/africafoundation/wp-content/uploads/sites/6/sites/6/HEALTH_nurse_blood-pressure-e1627651075100-1024x576.jpg 1024w, https://tcs000000.local/africafoundation/wp-content/uploads/sites/6/sites/6/HEALTH_nurse_blood-pressure-e1627651075100-768x432.jpg 768w, https://tcs000000.local/africafoundation/wp-content/uploads/sites/6/sites/6/HEALTH_nurse_blood-pressure-e1627651075100-1536x864.jpg 1536w, https://tcs000000.local/africafoundation/wp-content/uploads/sites/6/sites/6/HEALTH_nurse_blood-pressure-e1627651075100-300x169.jpg 1920w

notice the double **site/6/site6/**

After digging around, in your plugin file: **class-media.php** line 1712 there is a function **calculate_image_srcset_meta**
In that function you parse the image src, and thats the problem. 

Before the parse_url, wordpress sets the $image_meta['file'] as 
            **[file] => About-us_Bheki-and-ladies-laugh.jpg**
and after parse_url its
            **[file] => /africafoundation/wp-content/uploads/sites/6/About-us_Bheki-and-ladies-laugh.jpg**

So I think the workround is the image src should only be parsed if it's a cloudinary synced file.

I used the standard 2021 theme with only your plugin enabled and was able to replicate by printing the code below in a template file

```
	$id = get_post_thumbnail_id();
	$img = wp_get_attachment_image_srcset($id);

```

Hope this helps debug / fix the issue

Thanks
Dominic da Costa

**Comment by dominicpedro** [comment]
Hi Sorry, yes the line number is different, when I was debugging I edited the file my autoformatter affected the page, so lines were messed up.

In saying that it is the the latest version, v3.0.2, and I understand you have a check in place, but thats only when generating the image html. If I call the function directly,  you just have a simple filter hook that triggers and there is no check in that filter, so it will/does run. I think there needs to be a check within calculate_image_srcset_meta that doesnt apply the filename if its not synced.

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/1935639/160069639-8603cb88-7905-4a80-83fc-74bf4c99a19b.png">

**Comment by Vdeub-cloudinary** [comment]
Hi @dominicpedro,
My developer tested it on a vanilla Multisite, with a subdirectory and as you can see on the image below, there is no duplication.
![image](https://user-images.githubusercontent.com/66945106/160106539-ee7964e4-f090-461e-9acd-68e790ab836c.png)
Would you mind sharing a system report as documented [here](https://support.cloudinary.com/hc/en-us/articles/360022260599-How-to-generate-a-system-report-on-the-WordPress-plugin-)?
Thanks!

**Comment by Vdeub-cloudinary** [comment]
Hi @dominicpedro, we managed to reproduce it on the REST API answer - we will look further into it.
I'll keep you posted on our progress.

**Comment by Vdeub-cloudinary** [comment]
Hi @dominicpedro, we have just released `3.0.3`, could you please upgrade the plugin and see if that fixes your issue?
